.github/workflows/check.yml

```
@@ -32,44 +31,84 @@ jobs:
         - 3.8
         - 3.7
         - 3.6
+        - 3.5
+        - pypy3
         - 2.7
         - pypy2
-        - pypy3
-
+        include:
+        - {os: MacOs, py: brew@py3}
+        - {os: MacOs, py: brew@py2}
     steps:
-    - uses: actions/checkout@v2
-    - uses: tox-dev/action@v1.0.0
-      with:
-        python-version: ${{ matrix.py }}
-
+    - name: install OS dependencies
+      run:   |
+        ${{ runner.os == 'Linux' && 'sudo apt-get update -y && sudo apt-get install fish csh' || true}}
+        ${{ runner.os == 'macOS' && 'brew update -vvv && brew install fish tcsh' || true}}
+        ${{ runner.os == 'Windows' && ( matrix.py == '2.7' || matrix.py == 'pypy2' ) && 'choco install vcpython27 --yes -f' || true }}
+      shell: bash
     - name: setup python for tox
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
     - name: install tox
       run: python -m pip install tox
+    - uses: actions/checkout@v2
+    - name: use local virtualenv for tox
+      run: python -m pip install ${{ runner.os == 'Windows' && matrix.py == 'pypy3' && 'virtualenv<20' || '.'}}
     - name: setup python for test ${{ matrix.py }}
+      if: "!startsWith(matrix.py,'brew@py')"
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.py }}
-    - name: setup test suite for ${{ matrix.os }}-${{ matrix.py }}
-      run: tox -vve py --notest
-    - name: run test suite for ${{ matrix.os }}-${{ matrix.py }}
-      run: tox -e py
+    - name: load python from ${{ matrix.py }}
+      if: startsWith(matrix.py,'brew@py')
+      run: |
+        import os; import subprocess
+        version = "${{matrix.py}}"[-1]
+        subprocess.check_call(["brew", "install", f"python@{version}"])
+        print('::set-env name=PATH::' + (":".join([f"/usr/local/opt/python@{version}"] + os.environ["PATH"].split(":"))))
```

Using `add-path` command would be a bit more elegant https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path
```suggestion
        print(f'::add-path::/usr/local/opt/python@{version}')
```

---

Oh did not knew about it, will change this. Is this prepend path or add? I really need prepend here 🤔

---

Yep, if you click on that link above, here's the explanation you'll see:

> Prepends a directory to the system `PATH` variable for all subsequent actions in the current job. The currently running action cannot access the new path variable.